README.md

```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
 
 ```javascript
 // with the classic require...
-let Kuzzle = require('kuzzle-sdk/dist/kuzzle.js')
+const { Kuzzle } = require('kuzzle-sdk/dist/kuzzle.js')
 // ... or with the new import directive.
-import Kuzzle from 'kuzzle-sdk/dist/kuzzle.js'
+import { Kuzzle } from 'kuzzle-sdk/dist/kuzzle.js'
+```
+
+## Protocols used
+Actuellement, le SDK supporte nativement 3 protocols: `http`, `websocket` et `socketio`.  
+
+The main reason behind this is that while Socket.IO offers better compatibility with older web browsers, our raw WebSocket implementation is about 20% faster
+
+Which protocol is used when you connect to Kuzzle depends on multiple factors:
+
+#### NodeJS
+
+The protocol used is always raw WebSocket for `websocket`.
+
+#### Web Browsers
+
+If you choose `websocket`, the SDK will first try to use raw WebSocket to connect to Kuzzle. If the web browser does not support this protocol, then the SDK falls back to Socket.IO.
+
+## SDK Documentation
+
+### Connection to Kuzzle
+
+By default, the SDK is not connected to Kuzzle when it is instantiated. You must manually call the `kuzzle.connect()` method before using the SDK.  
+
+It is then possible to interact with the Kuzzle API through the SDK once the Promise returned by `kuzzle.connect()` has been resolved.  
+
+```js
+// Without async/await
+kuzzle.connect()
+  .then(() => {
+    // You are now connected to your Kuzzle instance.
+    return kuzzle.server.now();
+  })
+  .then(serverTime => console.log(serverTime))
+  .catch(error => console.error(error));
+
+// With async/await
+try {
+  await kuzzle.connect();
+  const serverTime = await kuzzle.server.now();
+
+  console.log(serverTime);
+} catch (error) {
+  console.error(error);
+}
+```
+
+### ISO with the Kuzzle API
+
+The version 6 of this SDK involve a massive refactor of the SDK structure to match the [Kuzzle API](https://docs.kuzzle.io/api-documentation/connecting-to-kuzzle/).  
+
+Each controller is accessible from the Kuzzle object. The controller's actions are named in the same way as in the API.  
+
+For example, for the `create` action of the `document` controller ([document#create](https://docs.kuzzle.io/api-documentation/controller-document/create)):
+```js
+const options = { refresh: 'wait_for' };
+const documentBody = { hello: 'world' };
+kuzzle.document.create('my-index', 'my-collection', 'my-uniq-id', documentBody, options)
+```
+
+The parameters of each method differ according to the parameters expected in the API.  
+To get the details of the parameters of each method, it is necessary for the moment to see the code of each controller on [Github](https://github.com/kuzzleio/sdk-javascript/tree/6-beta/src/controllers).
+
+### Promise based
+
+Each SDK method returns a Promise resolving on the result of the API return (the `result` property described in the API documentation).  
```

Comment:
All SDK methods return a promise resolving the `result` part of Kuzzle API responses. If an error occurs, the promise is rejected with an `Error` object embedding the `error` part of the API response.